source/Octopus.Tentacle/Commands/PollCommand.cs

```diff
@@ -60,38 +64,60 @@ protected override void Start()
 
         async Task StartAsync()
         {
+            if (!reuseThumbprint)
+            {
+                api.Validate();
+            }
+
             if (!string.IsNullOrEmpty(serverWebSocketAddress) && !string.IsNullOrEmpty(serverCommsAddress))
                 throw new ControlledFailureException("Please specify a --server-web-socket, or a --server-comms-address - not both.");
 
             var serverAddress = GetAddress();
 
+            var serverConfiguration = reuseThumbprint ?
+                CreateServerConfigurationFromFirstTrustedServer(serverAddress) :
+                await CreateServerConfigurationViaServerAPI(serverAddress);
+
+            configuration.Value.AddOrUpdateTrustedOctopusServer(serverConfiguration);
+            VoteForRestart();
+
+            log.Info("Polling endpoint configured");
+        }
+
+        OctopusServerConfiguration CreateServerConfigurationFromFirstTrustedServer(Uri serverAddress)
+        {
+            var firstTrustedServer = configuration.Value.TrustedOctopusServers.First();
```

Comment:
Is this guaranteed to be what the user expects as the first server? i.e. it is the same as either the first server in the list of `server-comms-addresses`

Related question, what happens when the `server-comms-address` was also supplied along with the list of extra comms addresses. Does the "first" trusted server instance refer to the server from the singular server-comms-address because that is preferred in the configure-and-run.sh script?

Reply:
There doesn't seem to be any kind of sorting done to the TrustedServers so the first Server should be the one that is registered first which should be the one that's used when registering the agent with OctopusServer.

If `ServerCommsAddress` (singular) is provided it's used for the initial registration with Server (eg: when the tentacle registers itself as a deployment target with server). If it's not provided, the first entry in `ServerCommsAddresses` (plural) is used. The left over entries in `ServerCommsAddresses` are registered as additional addresses.

So if `ServerCommsAddress` is provided, the first server will be that one. If it's not the first server will be `ServerCommsAddresses[0]`.

Does that make sense?

Reply:
Yea that makes sense. Just wondering if it will be clear to the user 🤔

Reply:
I'm considering deprecating the singular form and we can remove it in the next major release of the helm chart 🤔